lib/ClangImporter/ImporterImpl.h

```diff
@@ -651,6 +659,10 @@ class LLVM_LIBRARY_VISIBILITY ClangImporter::Implementation
     return Instance->getCodeGenOpts();
   }
 
+  importer::ClangSourceBufferImporter &getBufferImporterForDiagnostics() {
```

Comment:
The reason I'm so noisy about "for diagnostics" in this patch is because once upon a time we thought we'd map *all* source locations over whenever we imported a decl, so that notes like "superclass declared here" would point into the Objective-C sources. We steered away from that in the Swift 1 days partly because we ran out of time and partly because we didn't want people to immediately be dumped back in ObjC-land, so we went with the pretty-printed declarations instead. (Unfortunately, that gets completely lost when working in Xcode instead of on the command line.)

The other reason to be cagey about it today is that mapping a source location over isn't necessarily cheap—the Clang source manager needs to be kept alive to keep the Clang buffer alive (to avoid copying it), and the pretty source location is calculated up front and added to the Swift SourceManager like `#sourceLocation`. If we were mapping everything, I'd want to at least be more lazy about calculating the pretty source locations.

I don't think we'd ever want two separate buffer importers; it's more a separate object to avoid passing the ImporterImpl around in even more places.